### PR TITLE
Login and signup pages styled to match mockups

### PR DESCRIPTION
**Pull request by simalicrum**
### What this PR does (required):
- Login and signup pages styled to match mockups

### Screenshots / Videos (front-end only):
image

### Any information needed to test this feature (required):
- working mongodb required

### CODE REVIEW

File: client/src/components/DemoButton/DemoButton.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+import useStyles from './useStyles';
+import login from '../../helpers/APICalls/login';
+import { useAuth } from '../../context/useAuthContext';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { CircularProgress } from '@material-ui/core';
+
+const demoUser = { email: 'user@demo.com', password: 'password' };
```
Comment (rajivtitus) [inline client/src/components/DemoButton/DemoButton.tsx:10]: 👍🏽

Comment (rajivtitus) [inline client/src/components/DemoButton/DemoButton.tsx:10]: I think this feature was already a part of a previous PR which has been approved? 🤔 

When you have some time to spare, I would highly recommend creating a new repo and practice creating PRs - branching off of other branches - rebasing branches - et cetera.

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 2]
```diff
@@ -40,7 +41,7 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
         username: '',
       }}
       validationSchema={Yup.object().shape({
-        username: Yup.string().required('Username is required').max(40, 'Username is too long'),
+        username: Yup.string().required('Name is required').max(40, 'Name is too long'),
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:44]:
```suggestion
        name: Yup.string().required('Name is required').max(40, 'Name is too long'),
```

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 3]
```diff
@@ -40,7 +41,7 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
         username: '',
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:41]:
```suggestion
        name: '',
```

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 4]
```diff
@@ -51,66 +52,60 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
     >
       {({ handleSubmit, handleChange, values, touched, errors, isSubmitting }) => (
         <form onSubmit={handleSubmit} className={classes.form} noValidate>
+          <label htmlFor="username">
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:55]:
```suggestion
          <label htmlFor="name">
```

Ditto throughout!

**Review by rajivtitus — APPROVED** [review]
Nice work! There are some minor suggestions, please address them prior to merging. Thanks 👍🏽

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx (hunk 4 above)
Comment (simalicrum) [reply client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:55]: For consistency I've changed the text field and label to 'Username' and placeholder to 'Create a username' then username is disaggregated from first and last name which is more typical of login interfaces.

**Comment by simalicrum** [comment]
I've moved the shared UI elements from the login and signup into ModalButton and ModalPaper function components for easy of styling and future reuse. Also I fixed a couple of outstanding CSS issues.

**Pull request merged**